rfcs/convergence/first-rule-of-aria.md

```diff
@@ -0,0 +1,204 @@
+    <tr>
+      <th colspan="2">The table header</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>The table body</td>
+      <td>with two columns</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+Since table is a common example of virtualization scenarios, where elements, content, headers, etc,. can be added on the go by JS code, the structural behavior provided by semantic elements can be easily compromised. For example in the case where a div is added between `tbody` and `tr` for scrolling purposes.
+
+```tsx
+<Table>
+  <TableHead>
+    <TableRow>
+      <TableCell colSpan={2}>
+        The table header
+      </TableCell>
+    </TableRow>
+  </TableHead>
+  <TableBody>
+    <div>
+      <TableRow>
+        <TableCell>
+          The table body
+        </TableCell>
+        <TableCell>
+          with two columns
+        </TableCell>
+      </TableRow>
+    </div>
+  </TableBody>
+</Table>
+
+<table>
+  <thead>
+    <tr>
+      <th colspan="2">The table header</th>
+    </tr>
+  </thead>
+  <tbody>
+    <!-- This div breaks the markup, since tbody cannot contain  -->
+    <div>
+      <tr>
+        <td>The table body</td>
+        <td>with two columns</td>
+      </tr>
+    </div>
+  </tbody>
+</table>
+```
+
+The idea of going for semantic first approach is to ensure that in the default cases, where the user of the lib simply doesn't care about edge cases such as `a` element inside of a `button` element, semantic elements will be used ensuring a better overall experience across different environments, and in the edge cases there's an option to dispose the semantic element for something more generic and ARIA.
+
+The example above could easily be converted to ARIA:
+
+```tsx
+// This as="div" is informs all compound components to opt-out of semantic-first approach
+<Table as="div">
+  <TableHead>
+    <TableRow>
+      <TableCell colSpan={2}>
+        The table header
+      </TableCell>
+    </TableRow>
+  </TableHead>
+  <TableBody>
+    <div>
+      <TableRow>
+        <TableCell>
+          The table body
+        </TableCell>
+        <TableCell>
+          with two columns
+        </TableCell>
+      </TableRow>
+    </div>
+  </TableBody>
+</Table>
+
+<div role="table">
+  <div role="rowgroup">
+    <div role="row">
+      <span role="columnheader" aria-colspan="2">The table header</span>
+    </div >
+  </div>
+  <div role="rowgroup">
+    <div>
+      <div role="row">
+        <span role="cell">The table body</span>
+        <span role="cell">with two columns</span>
+      </div>
+    </div>
+  </div>
+</div>
+```
+
+This proposal does not introduce any API, is just a pattern to be followed when creating converged components, and valid to add that in extreme cases, like the table, that semantic first approach is simply not viable, than this should be ignored.
+
+### Pros and Cons
+
+#### Pros
+
+- Better support overall for A11y
+
+#### Cons
+
+- Styles override to remove styles from semantic elements will be necessary
```

Comment:
This is a con for the fact that if we don't use semantic elements by default we simply don't need to override them, since `div` have not much to be overridden.